modules/subscription/main.tf

```diff
@@ -122,3 +122,27 @@ resource "azapi_resource_action" "subscription_cancel" {
   ]
 }
 
+resource "azapi_resource" "subscription_dfc_contact" {
+  count     = var.subscription_dfc_contact_enabled ? 1 : 0
+  parent_id = "/subscriptions/${local.subscription_id}"
+  type      = "Microsoft.Security/securityContacts@2020-01-01-preview"
+  name      = "default" //  The only valid name for security contact is 'default'
+
+  body = jsonencode({
+    properties = {
+      emails = var.subscription_dfc_contact.emails
+      phone  = var.subscription_dfc_contact.phone
+
+      alertNotifications = {
+        state           = var.subscription_dfc_contact.alert_notifications == "Off" ? var.subscription_dfc_contact.alert_notifications : "On"
+        minimalSeverity = var.subscription_dfc_contact.alert_notifications == "Off" ? "Low" : var.subscription_dfc_contact.alert_notifications
+      }
+
+      // Either an email address or at least one role must be set to receive notification alerts.
+      notificationsByRole = {
+        state = var.subscription_dfc_contact.emails == "" || length(var.subscription_dfc_contact.notifications_by_role) > 0 ? "On" : "Off"
```

Comment:
for more clarity, maybe use the same logic for the two line 143 and 144
roles = var.subscription_dfc_contact.emails == "" && length(var.subscription_dfc_contact.notifications_by_role) == 0 ? ...

Reply:
Hi Christian12cc, thanks for the feedback. Based on the logic/truth table below, do you think it would it be clearer if we changed line 143 to an AND logic?

```hcl
state = var.subscription_dfc_contact.emails != "" && length(var.subscription_dfc_contact.notifications_by_role) == 0 ? "Off" : "On"
```

| var.emails    | var.notifications_by_role | _state_               | _roles_    |
|-----------|-----------------------|---------------------|----------|
| Empty     | Empty                 | On                  | [Owner]  |
| Empty     | Provided              | On                  | Provided |
| Provided  | Empty                 | Off                 | Empty    |
| Provided  | Provided              | On                  | Provided |

Reply:
Hi Felipe
I have a look to the API doc (https://learn.microsoft.com/en-us/azure/templates/microsoft.security/securitycontacts?pivots=deployment-language-terraform) .
I think the state in the "notificationsByRole" is only for the RBAC role. So in fact you don't need to test the email adsress

<!--StartFragment-->
state in  NotificationsByRole | Defines whether to send email notifications from AMicrosoft Defender for Cloud to persons with specific RBAC roles on the subscription. | "Off""On"
-- | -- | --


<!--EndFragment-->


and I think that for the state (line 137) in AlertNotifications, we should test that an email address is defined.

<!--StartFragment-->
state in AlertNotifications | Defines if email notifications will be sent about new security alerts | "Off""On"
-- | -- | --


<!--EndFragment-->

Reply:
Hi Christian,

Either an email address or at least one role must be provided to enable alert notifications. If neither is provided, you will get "Error: AlertNotifications is enabled without any SecurityContact info". To avoid getting this error, I was setting the notificationsByRole to "Owner" if no email was provided.

Seems like it is better to let the API handle this situation, keep it simple, and avoid confusion. I will remove the conditional logic from the module, while noting this requirement in the module documentation.

Thanks

Reply:
Perfect
Tanks Felipe